modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl

```diff
@@ -1689,11 +1691,46 @@ public abstract class Base${schemaName}ResourceTestCase {
 		}
 	</#list>
 
-	protected List<GraphQLField> getGraphQLFields() {
+	protected List<GraphQLField> getGraphQLFields() throws Exception {
 		List<GraphQLField> graphQLFields = new ArrayList<>();
 
-		for (String additionalAssertFieldName : getAdditionalAssertFieldNames()) {
-			graphQLFields.add(new GraphQLField(additionalAssertFieldName));
+		for (Field field : ReflectionUtil.getDeclaredFields(
+				${configYAML.apiPackagePath}.dto.${escapedVersion}.${schemaName}.class)) {
+
+			if (!ArrayUtil.contains(getAdditionalAssertFieldNames(), field.getName())){
+				continue;
+			}
+
+			graphQLFields.addAll(getGraphQLFields(field));
+		}
+
+		return graphQLFields;
+	}
+
+	protected List<GraphQLField> getGraphQLFields(Field... fields) throws Exception {
+		List<GraphQLField> graphQLFields = new ArrayList<>();
+
+		for (Field field : fields) {
+			com.liferay.portal.vulcan.graphql.annotation.GraphQLField
+				graphQLField = field.getAnnotation(
+					com.liferay.portal.vulcan.graphql.annotation.GraphQLField.
+						class);
+
+			if (graphQLField != null) {
+				Class<?> type = field.getType();
+
+				if (type.isArray()) {
```

Comment:
@inacionery I am not sure why you only care about array here, what about other collection types like List, Set and so on? 

Tina.

Reply:
The generator uses only array